### ISSUE TITLE
[false-positive] Private field “overriding” super’s private field

### ISSUE THREAD

**Issue by brandonkelly** [kalessil/phpinspectionsea#71]
Example:

```php
class A
{
    private $_foo;
}

class B extends A
{
    private $_foo;
}
```

This will trigger two inspection warnings on `B::$_foo`:

1. Field ‘_foo’ is already defined in A. Consider initializing it in a constructor or making static.
2. Likely needs to be protected (also in the parent class).

Event though (1) is technically correct for saying that `_foo` is already defined in the the super class, it’s irrelevant here because private properties are not accessible to subclasses. And (2) is implying that `A::$_foo` is protected, which is not true. So neither of these warnings should be present.

**Comment by kalessil** [comment]
@brandonkelly : this fixed, will be available in the next release (~Feb 2017). You'll need to deactivate reporting private fields reporting in settings.

Additionally I added [documentation](https://github.com/kalessil/phpinspectionsea/blob/master/docs/architecture.md#class-overrides-a-field-of-a-parent-class) explaining possible options for working warning out. Hope this will be useful.
